ui/shared/utils/panelAppUtils.ts

```diff
@@ -74,15 +75,27 @@ export type PanelAppItem = {
  * and concatenated item displays as values.
  *
  * @param {PanelAppItem[] | null | undefined} items - The array of PanelApp items to format.
+ * @param {string[] | null | undefined} selectedMOIs - Optional array of selected MOIs for filtering.
  * @returns {Record<string, string> | never[]} A record where keys represent confidence levels and values
  * are concatenated item displays, or an empty array if `items` is falsy.
  */
-export const formatPanelAppItems = (items: PanelAppItem[] | null | undefined): Record<string, string> | never[] => {
+export const formatPanelAppItems = (
+  items: PanelAppItem[] | null | undefined,
+  selectedMOIs?: string[] | null
+): Record<string, string> | never[] => {
   if (!items) {
     return []
   }
 
-  return items.reduce((acc, item) => {
+  const filteredItems = items.filter((item) => {
+    if (!selectedMOIs || selectedMOIs.length === 0) {
+      return true
+    }
+    const initials = moiToMoiInitials(item.pagene?.modeOfInheritance, false)
+    return selectedMOIs.some((moi) => initials.includes(moi))
+  })
+
+  return filteredItems.reduce((acc, item) => {
```

Comment:
this would be slightly neater as a single `reduce` with a helper method
```suggestion
  const hasSelectedMoi = (item, selectedMOIs) => {
    if (!selectedMOIs || selectedMOIs.length === 0) {
      return true
    }
    const initials = moiToMoiInitials(item.pagene?.modeOfInheritance, false)
    return selectedMOIs.some((moi) => initials.includes(moi))
  }
 ...
 
  return items.reduce((acc, item) => {
    if (!hasSelectedMoi(item, selectedMOIs) {
      return acc
    }
    ...
```